cpp/src/Slice/SliceUtil.cpp

```diff
@@ -470,59 +470,49 @@ Slice::checkIdentifier(const string& id)
         name = id;
     }
 
+    // Validity is only determined by invalid syntax that triggers errors, not warnings.
     bool isValid = true;
+
     // check the identifier for reserved suffixes
     static const string suffixBlacklist[] = { "Helper", "Holder", "Prx", "Ptr" };
     for(size_t i = 0; i < sizeof(suffixBlacklist) / sizeof(*suffixBlacklist); ++i)
     {
         if(name.find(suffixBlacklist[i], name.size() - suffixBlacklist[i].size()) != string::npos)
         {
-            unit->error("illegal identifier `" + name + "': `" + suffixBlacklist[i] + "' suffix is reserved");
-            isValid = false;
+            unit->warning(ReservedIdentifiers, "`" + suffixBlacklist[i] +
```

Comment:
_the_ `Prx` suffix is reserved by Ice?

Reply:
I originally added a 'the' at the start, but after adding it, the message sounded weird and repetitive from having two `the`s that close together:
`the Helper suffix is reserved by Ice: the identifier 'foo' may conflict with generated code`

We originally had `...: 'Helper' suffix is reserved`, so I decided to follow it and drop the first to make it sound better.

Reply:
What about:
identifiers with a `Prx` suffix are reserved by Ice: using `FooPrx` as an identifier may result in name collisions in the generated code

Reply:
A little wordy, but I like it! Fixed.